### PR TITLE
Comment

### LINKED ISSUE #10: Thêm comment cho post
Thêm phần comment cho mỗi post.
Design sẽ giống như facebook. Bên dưới post hiển thị các comment cũ và chỗ nhập comment mới + hiển thị reaction.
Em sẽ phải tự tìm hiểu, tách thành các task con.
Đầu tiên là tách thành các task con để a review rồi sẽ làm tiếp

### PR DESCRIPTION
**Pull request by NishaLab**
Issue: #10 
Real Time: 6 hours
image

### CODE REVIEW

File: app/models/micropost.rb [hunk 1]
```diff
@@ -3,9 +3,11 @@
 # simple micropost class has 1 attribute: content
 class Micropost < ApplicationRecord
   belongs_to :user
+  belongs_to :parent, class_name: "Micropost", optional: true
```
Comment (linuxhjkaru) [inline app/models/micropost.rb]: nên đặt tên là parent_post

File: app/models/micropost.rb [hunk 2]
```diff
@@ -3,9 +3,11 @@
...
   has_one_attached :image
   has_many :reactions, dependent: :destroy
-
+  has_many :childs, class_name: "Micropost",
```
Comment (linuxhjkaru) [inline app/models/micropost.rb]: nên là child_posts

File: app/views/microposts/_micropost.html.erb [hunk 3]
```diff
@@ -9,9 +9,24 @@
...
+    </section>
   </div>
+  <div class="row comment_form">
+    <section>
+      <%= render 'shared/micropost_form', parent_id: micropost.id %>
+    </section>
+  </div>
+  <div class= "row comment">
+    <section id = "comment-<%= micropost.id%>">
+      <% for micropost in micropost.childs%>
```
Comment (linuxhjkaru) [inline app/views/microposts/_micropost.html.erb]: dùng ` micropost.childs.each do |child_micropost|`

**Review by NishaLab — COMMENTED** [review]
fix bug when in user page instead of create comment, page will update the parent post 
reason: wrong model for form_with helper cause form_to helper automatically change to PATCH request instead of POST

**Pull request merged**